core/src/main/java/org/apache/iceberg/PartitionSummary.java

```diff
@@ -43,7 +43,7 @@
   }
 
   List<PartitionFieldSummary> summaries() {
-    return Lists.transform(Arrays.asList(fields), PartitionFieldStats::toSummary);
+    return Arrays.stream(fields).map(PartitionFieldStats::toSummary).collect(Collectors.toList());
```

Comment:
This transformation has to be eager as `PartitionFieldStats` is not serializable.